PublishedModules/glueckself/mpdclient/mpdclient.py

```diff
@@ -0,0 +1,80 @@
+import mpdhelper
+
+from core.base.model.Intent import Intent
+from core.base.model.Module import Module
+from core.dialog.model.DialogSession import DialogSession
+from core.commons.commons import online
+
+class mpdclient(Module):
+	"""
+	Author: glueckself
+	Description: Control an MPD server
+	"""
+
+	_INTENT_PLAY = Intent('Play')
+	_INTENT_STOP = Intent('Stop')
+	_INTENT_NEXT = Intent('Next')
+	_INTENT_PREV = Intent('Prev')
+
+	def __init__(self):
+		self._SUPPORTED_INTENTS = [
+			self._INTENT_PLAY,
+			self._INTENT_STOP,
+			self._INTENT_NEXT,
+			self._INTENT_PREV
+		]
+
+		super().__init__(self._SUPPORTED_INTENTS)
+		
+		config = self.getConfig('mpdclient')
+		self._host = config.get('mpdHost')
+		self._port = config.get('mpdPort')
+		self._password = config.get('mpdPassword')
+		
+		self._mpdConnected = False
+		
+		if self._host:
+			self._connect()
+		else:
+			self._logger.warn(f'[{self.name}] MPD host not configured, not doing anything.')
+		
+	def _connect(self):
+		self._mpd = mpdhelper.MPDClient()
+		try:
+			self._mpd.connect(self._host, self._port, 5)
+		except:
+			self._logger.warn(f'[{self.name}] Failed to connect to mpd host {self._host}:{self._port}, retrying in 10s.')
+			self.ThreadManager.doLater(interval=10, func=self._connect)
+		
+		if self._password:
+			self._mpd.password(self._password)
+			
+		self._logger.info(f'[{self.name}] Connected to mpd host {self._host}:{self._port}')
+		self._mpdConnected=True
```

Comment:
when self._mpd.connect throws an exception self._mpdConnected gets set to true which appears unwanted

Reply:
Fixed, thank you.